### PR TITLE
Message window, Remove token button and inactivate sync button

### PR DESCRIPTION
**Pull request by ModeenF**
I thought it was to little text on whats going on. Yes you can run in terminal but as ordinary user basic information are good to show.

As I wanted to re test I needed to remove my token.. so I made a remove token button..

I also made sync button and Remve button inactivateed on sync.

In the diff the _Sync() looks vierd..
I remove all of the "_Sync()" And had added "_SetStatusMessage(const char* str)"

### CODE REVIEW

**Comment by ModeenF** [comment]
I have alot of small changes.. How can I remove them and still have the changes?

**Comment by bitigchi** [comment]
You need to use feature branches.

**Comment by ModeenF** [comment]
New to Git.  But for all changes or are we talking about this pull request?

**Comment by bitigchi** [comment]
Yes, for each pull request you ought to use feature branches, so that the changes do not get intertwined. Once you branch off of a clean master, everything should be okay.

**Comment by humdingerb** [comment]
For feature branches, see https://github.com/haikuports/haikuports/wiki/DevelopmentModel .

File: src/plugin/GoogleCalendar/EventSyncWindow.cpp [hunk 1]
```diff
@@ -98,13 +107,17 @@ EventSyncWindow::_InitInterface()
 	BView* fMainView = new BView("MainView", B_WILL_DRAW);
 	fMainView->SetViewUIColor(B_PANEL_BACKGROUND_COLOR);
 
-	fStatusLabel = new BStringView("Status Label",
-		B_TRANSLATE("No Data Available"));
+	fStatusLabel = new BStringView("Status Label","");
 	fSyncButton = new BButton(NULL, "Sync", new BMessage(kSyncPressed));
```
Comment (humdingerb) [inline src/plugin/GoogleCalendar/EventSyncWindow.cpp:111]:
```suggestion
	fSyncButton = new BButton(NULL, B_TRANSLATE("Synchronize"), new BMessage(kSyncPressed));
```

File: src/plugin/GoogleCalendar/EventSyncWindow.cpp [hunk 2]
```diff
@@ -98,13 +107,17 @@ EventSyncWindow::_InitInterface()
...
+	fRemoveButton = new BButton(NULL, "Remove Token", new BMessage(kRemovePressed));
```
Comment (humdingerb) [inline src/plugin/GoogleCalendar/EventSyncWindow.cpp:112]: Again, add the B_TRANSLATE() macro, also sentence casing "Remove token".
However, that is quite technical. What does it do in layman's terms? "Re-download events", "Reset calender", etc. ?

File: src/plugin/GoogleCalendar/EventSyncWindow.cpp [hunk 3]
```diff
@@ -40,6 +43,8 @@ EventSyncWindow::EventSyncWindow()
 		syncDataDir.CreateDirectory(syncDataPath.Path(), &syncDataDir);
 	}
 
+	fLogMessage = new BTextView("Log");
```
Comment (humdingerb) [inline src/plugin/GoogleCalendar/EventSyncWindow.cpp:46]: This should probably be in a BScrollView.

File: src/plugin/GoogleCalendar/EventSyncWindow.cpp [hunk 4]
```diff
@@ -182,20 +205,16 @@ EventSyncWindow::_LoadSyncData()
 			_SetStatusLabel(lastSyncStatus, *lastSyncTime);
 		}
 	}
-
 	else
 	{
-		BAlert* alert  = new BAlert(B_TRANSLATE("Error"),
-			B_TRANSLATE("There was an error in loading the last sync data."),
-			NULL, B_TRANSLATE("OK"),NULL, B_WIDTH_AS_USUAL, B_WARNING_ALERT);
+		_SetStatusMessage(B_TRANSLATE("There was an error in loading the last sync data."));
```
Comment (humdingerb) [inline src/plugin/GoogleCalendar/EventSyncWindow.cpp:210]: As the window width is limited (maybe make it resizable?), status messages should be short.

```suggestion
		_SetStatusMessage(B_TRANSLATE("Error loading last sync data."));
```

File: src/plugin/GoogleCalendar/EventSyncWindow.cpp [hunk 5]
```diff
@@ -206,22 +225,41 @@ EventSyncWindow::_SaveSyncData(bool status)
 	time_t syncTime = time(NULL);
 
 	if (file->InitCheck() == B_OK) {
+		_SetStatusMessage(B_TRANSLATE("SaveSyncData"));
```
Comment (humdingerb) [inline src/plugin/GoogleCalendar/EventSyncWindow.cpp:228]: I think we shouldn't mix end user and developer terms.
```suggestion
		_SetStatusMessage(B_TRANSLATE("Saving sync data"));
```

File: src/plugin/GoogleCalendar/EventSyncWindow.cpp [hunk 6]
```diff
@@ -206,22 +225,41 @@ EventSyncWindow::_SaveSyncData(bool status)
...
 		message->Flatten(file);
 	}
-
 	else
 	{
-		BAlert* alert  = new BAlert(B_TRANSLATE("Error"),
-			B_TRANSLATE("There was an error in saving the recent sync data."),
-			NULL, B_TRANSLATE("OK"),NULL, B_WIDTH_AS_USUAL, B_WARNING_ALERT);
-
+		_SetStatusMessage(B_TRANSLATE("There was an error in saving the recent sync data."));
```
Comment (humdingerb) [inline src/plugin/GoogleCalendar/EventSyncWindow.cpp:236]:
```suggestion
		_SetStatusMessage(B_TRANSLATE("Error saving recent sync data."));
```

File: src/plugin/GoogleCalendar/EventSyncWindow.cpp [hunk 7]
```diff
@@ -206,22 +225,41 @@ EventSyncWindow::_SaveSyncData(bool status)
...
+	BPasswordKey key;
+	BKeyStore keyStore;
+	if(keyStore.GetKey(kAppName, B_KEY_TYPE_PASSWORD, "refresh_token", key) == B_OK) {
+		status_t result = keyStore.RemoveKey(kAppName, key);
+		if (result != B_OK) {
+			_SetStatusMessage(B_TRANSLATE("Token removed"));
+		}
+		else
+		{
+			_SetStatusMessage(B_TRANSLATE("Error remove token"));
```
Comment (humdingerb) [inline src/plugin/GoogleCalendar/EventSyncWindow.cpp:258]:
```suggestion
			_SetStatusMessage(B_TRANSLATE("Error removing token"));
```

**Review by humdingerb — COMMENTED** [review]
The log is quite technical, but ver helpful to provide troubleshooting info.
Maybe it should be hidden with a "Show log" checkbox. Or an "expander" arrow, though I'm not sure where the API provides those or what those widgets are called. They may have fallen out of style anyway, because they've been removed in the places I remember (the Find and Info panels)...

**Comment by scottmc** [comment]
What's the status here?

**Comment by scottmc** [comment]
Closing this.  Let's just remove the menu option so the user doesn't see the broken parts but future developer has a point to start from.
